authors/apps/profiles/test/test_user_profile.py

```diff
@@ -0,0 +1,93 @@
+            "last_name": "asiago",
+            "gender": "G",
+            "bio": "my bio"
+        }
+
+    def test_get_user_profile(self):
+        """Test get profile upon registrations"""
+        # register user
+        register = self.client.post(
+            self.signup_url, self.signup_data, format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+
+        # get user profile
+        profile = self.client.get(
+            reverse('profiles:put-profile', kwargs={'username': 'johndoe'}), format='json')
+        self.assertEqual(profile.status_code, status.HTTP_200_OK)
+
+    def test_updating_profile(self):
+        """Test post profile upon registrations"""
+        # sign up a user and get token
+        register = self.client.post(
+            self.signup_url, self.signup_data, format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        token = register.data['token']
+
+        # update profile
+        profile = self.client.put(reverse('profiles:put-profile', kwargs={'username': 'johndoe'}),
+                                  self.profile_data, format='json', HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(profile.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            profile.data, {'response': 'profile has been updated successfully '})
+
+    def test_updating_empty_first_name(self):
+        """Test post profile upon registrations"""
+        # sign up a user and get token
+        register = self.client.post(
+            self.signup_url, self.signup_data, format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        token = register.data['token']
+
+        # update profile
+        profile = self.client.put(reverse('profiles:put-profile', kwargs={'username': 'johndoe'}),
+                                  self.empty_first_name, format='json', HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(profile.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual({'errors': {'first_name': [
+                         'This field may not be blank.']}}, json.loads(profile.content))
+
+    def test_updating_invalid_gender(self):
+        """Test post profile upon registrations"""
+        # sign up a user and get token
+        register = self.client.post(
+            self.signup_url, self.signup_data, format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        token = register.data['token']
+
+        # update profile
+        profile = self.client.put(reverse('profiles:put-profile', kwargs={'username': 'johndoe'}),
+                                  self.invalid_gender, format='json', HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(profile.status_code, status.HTTP_400_BAD_REQUEST)
+        data = {'errors': {'gender': ['Please enter M if you are male, F if you are female or '
```

Comment:
Just define a message here i.e. `msg = 'Please enter M if you are male, F if you are female or ....` and then compare it with the response, instead of having all the dictionaries and a list.

Reply:
fixed it